blocksync/block_stats.go

```diff
@@ -0,0 +1,98 @@
+package blocksync
+
+// blockStats is a circular buffer that holds at most n elements
+// and provides O(1) average calculation and O(1) max retrieval
+type blockStats struct {
+	buffer   []float64
```

Comment:
can we rename this `buffer` to be `currentHeights` or similar to know what it contains?